### PR TITLE
Expose more methods in ExcessMapEstimator

### PR DESCRIPTION
**Pull request by QRemy**
Expose methods to compute kernel, mask, and reco_exposure in ExcessMapEstimator. This will be needed in an other PR to add support for estimation over datasets.

### CODE REVIEW

**Review by AtreyeeS — APPROVED** [review]
Thanks @QRemy ! Irrespective of how a joint ExcessMapEstimator is finally implemented, I think it is good to expose these functions

File: gammapy/estimators/map/excess.py [hunk 1]
```diff
@@ -195,42 +195,63 @@ def run(self, dataset):
...
 
+        Returns
+        -------
+        kernel : `~astropy.convolution.Tophat2DKernel`
+            Kernel
+        """
         pixel_size = np.mean(np.abs(dataset.counts.geom.wcs.wcs.cdelt))
         size = self.correlation_radius.deg / pixel_size
         kernel = Tophat2DKernel(size)
+        return kernel
```
Comment (adonath) [inline gammapy/estimators/map/excess.py:214]: For consistency with the `TSMapEstimator` I think we should return a `Map` object here.

File: gammapy/estimators/map/excess.py [hunk 2]
```diff
@@ -195,42 +195,63 @@ def run(self, dataset):
...
 
-        geom = dataset.counts.geom
+    def get_mask(self, dataset):
```
Comment (adonath) [inline gammapy/estimators/map/excess.py:216]: Again for consistency with the `TSMapEstimator`, just rename to `estimate_mask_default()`.

File: gammapy/estimators/map/excess.py [hunk 3]
```diff
@@ -195,42 +195,63 @@ def run(self, dataset):
...
             mask = dataset.mask_safe
         else:
-            mask = Map.from_geom(geom, data=True, dtype=bool)
+            mask = Map.from_geom(dataset.counts.geom, data=True, dtype=bool)
+        return mask
 
-        counts_stat = convolved_map_dataset_counts_statistics(
-            dataset, kernel, mask, self.correlate_off
-        )
+    def estimate_reco_exposure(self, dataset, kernel, mask):
```
Comment (adonath) [inline gammapy/estimators/map/excess.py:237]: Rename to `estimate_exposure_reco_energy` for consistency with the utility method.

**Review by adonath — COMMENTED** [review]
Thank @QRemy, I have left some comments to rename method for consistency within Gammapy.

**Review by adonath — APPROVED** [review]
Thanks a lot @QRemy, no further comments from my side. The nicely further improves the consistency between `ExcessMapEstimator` and `TSMapEstimator`. As this only introduces new methods there is no need to deprecate...

**Review by AtreyeeS — APPROVED** [review]
Thanks @QRemy !

**Comment by adonath** [comment]
Thanks @AtreyeeS for merging!

**Pull request merged**